src/ucp/core/ucp_mm.c

```diff
@@ -1012,7 +1013,8 @@ ucp_memh_alloc(ucp_context_h context, void *address, size_t length,
         goto err_dealloc;
     }
 
-    status = ucp_memh_init_all_uct_reg(context, memh, uct_flags, alloc_name);
+    status = ucp_memh_init_all_uct_reg(context, memh, uct_flags, ucp_flags,
```

Comment:
instead of passing ucp_memh_create_from_mem, IMO we need to make sure UCP_MEMH_FLAG_NO_RCACHE is present in memh->flags field, from ucp_memh_init(). then ucp_memh_init_uct_reg could check  memh->flags

Reply:
ok